test/NamingStrategy/MapNamingStrategyTest.php

```diff
@@ -9,50 +9,93 @@
 
 namespace ZendTest\Hydrator\NamingStrategy;
 
-use InvalidArgumentException;
 use PHPUnit\Framework\TestCase;
+use Zend\Hydrator\Exception;
 use Zend\Hydrator\NamingStrategy\MapNamingStrategy;
 
 /**
- * @covers Zend\Hydrator\NamingStrategy\MapNamingStrategy<extended>
+ * Tests for {@see MapNamingStrategy}
+ *
+ * @covers \Zend\Hydrator\NamingStrategy\MapNamingStrategy
  */
 class MapNamingStrategyTest extends TestCase
 {
-    public function testHydrateMap()
+    public function testConstructorRaisesExceptionIfNoExtractionOrHydrationMapProvided()
     {
-        $namingStrategy = new MapNamingStrategy(['foo' => 'bar']);
+        $this->expectException(Exception\InvalidArgumentException::class);
+        $this->expectExceptionMessage('requires one or both of an array $hydrationMap and array $extractionMap');
 
-        $this->assertEquals('bar', $namingStrategy->hydrate('foo'));
-        $this->assertEquals('foo', $namingStrategy->extract('bar'));
+        new MapNamingStrategy();
     }
 
-    public function testHydrateAndExtractMaps()
+    public function invalidMapValues() : iterable
```

Comment:
You also want to test with string keys. Yes, integer-ish keys are auto-vast to integers by the horrors of the PHP engine, but it is an interesting edge case to cover.